tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/CombinationListingFeatureContext.php

```diff
@@ -168,10 +212,21 @@ private function getDbField(string $field): string
      * @param string $productReference
      * @param array $dataRows
      * @param bool $wholeList if true then search criteria wont be applied
+     * @param string|null $shopReference if null uses default shopId
      */
-    private function assertCombinations(string $productReference, array $dataRows, bool $wholeList = false): void
-    {
-        $searchCriteriaKey = $this->getSearchCriteriaKey($productReference);
+    private function assertCombinations(
+        string $productReference,
+        array $dataRows,
+        bool $wholeList = false,
+        ?string $shopReference = null
```

Comment:
```suggestion
        int $shopId
```

Reply:
As I said in previous comments this is too sensitive to be left optional The purpose of adding shop constraint everywhere is to remove the magic auto fallback everywhere, including in the behat tests

So this parameter should be always required, and it's the responsibility of the calling code to provide a shopId, even if it means using `$this->getDefaultShopId()` as a fallback but at least it's clear from the start The fallback is not hidden